client/src/handlers/handlers.ts

```diff
@@ -260,3 +260,222 @@ export function makeHandleRunJobWithProgress (
+ * 
+ * @param context vscode.ExtensionContext
+ * 
+ * @param launchOptions LaunchOptions
+ * 
+ * @returns function handler
+ */
+export function cmdRunProject(
+    context: vscode.ExtensionContext, 
+    launchOptions: LaunchOptions = defaultLaunchOptions
+    ) {
+        return async function handler () {
+            const flixFilename = await getFlixFilename(context, launchOptions)
+            const cmd = ['java', '-jar', flixFilename, 'run']
+            let terminal = getTerminal('run')
+            passCommandToTerminal(cmd, terminal)
+        }
+}
+
+/**
+ * runs the benchmarks for the current project using command `java -jar flix.jar benchmark`
+ * 
+ * @param context vscode.ExtensionContext
+ * 
+ * @param launchOptions LaunchOptions
+ * 
+ * @returns function handler
+ */
+export function cmdBenchmark(
+    context: vscode.ExtensionContext, 
+    launchOptions: LaunchOptions = defaultLaunchOptions
+    ) {
+        return async function handler () {
+            const flixFilename = await getFlixFilename(context, launchOptions)
+            const cmd = ['java', '-jar', flixFilename, 'benchmark']
+            let terminal = getTerminal('benchmark')
+            passCommandToTerminal(cmd, terminal)
+        }
+}
+
+/**
+ * runs all the tests for the current project using command `java -jar flix.jar test`
+ * 
+ * @param context vscode.ExtensionContext
+ * 
+ * @param launchOptions LaunchOptions
+ * 
+ * @returns function handler
+ */
+export function cmdTests(
+    context: vscode.ExtensionContext, 
+    launchOptions: LaunchOptions = defaultLaunchOptions
+    ) {
+        return async function handler () {
+            const flixFilename = await getFlixFilename(context, launchOptions)
+            const cmd = ['java', '-jar', flixFilename, 'test']
+            let terminal = getTerminal('test')
+            passCommandToTerminal(cmd, terminal)
+        }
+}
+
+/**
+ * runs the custom tests for the current project using command `java -jar flix.jar test <test01> <test02> ...`
+ * 
+ * @param context vscode.ExtensionContext
+ * 
+ * @param launchOptions LaunchOptions
+ * 
+ * @returns function handler
+ */
+export function cmdTestWithFilter(
+    context: vscode.ExtensionContext, 
+    launchOptions: LaunchOptions = defaultLaunchOptions
+    ) {
+        return async function handler () {
+            const flixFilename = await getFlixFilename(context, launchOptions)
+            const cmd = ['java', '-jar', flixFilename, 'test']
+            const input = await vscode.window.showInputBox({
+                prompt: "Enter names of test functions separated by spaces",
+                placeHolder: "test01 test02 ...",
+                ignoreFocusOut: true
+            })
+            if(input != undefined)
+            {
+                cmd.push(...input.split(' '))
+                let terminal = getTerminal('testWithFilter')
+                passCommandToTerminal(cmd, terminal)
```

Comment:
Is it okay to pass the command in this format:
 `java -jar flix.jar test test01 test02 test03`
or the format should be (with apostrophes):
`java -jar flix.jar test 'test01 test02 test03'`

Reply:
Good question. I think it should be in one quote, because people may want to write e..g `*.foo*` which should NOT be expanded by the shell- so yes, it should be escaped.